lib/IR/IR.cpp

```diff
@@ -102,6 +102,26 @@ Instruction::Operand Instruction::getOperand(unsigned idx) const {
   return ops_[idx];
 }
 
+unsigned Instruction::getNumInputs() const {
```

Comment:
Yes this is intentional. The InOuts operands are counted twice (both as inputs and outputs). This is similar to the `instrument-debug` which instruments the InOut operands twice because these operands should be dumped before and after the execution of the instruction. Indeed numInputs + numOutputs can be greater than the effective number of operands but I guess the purpose of the callbacks is not necessarily to know the effective number of operands but to have the ability to dump all the required data. By inspecting the meta file `instrument-ir.info` you have this information because in that case an operand with same name is registered both as input and output.